core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -322,36 +363,149 @@ wasi_addr_ip_to_bh_ip_addr_buffer(__wasi_addr_ip_t *addr,
+                }
+            }
+        }
+    
+    #else
+    {
+        __wasi_clockid_t nclock_id;
+        if (!convert_clockid(clock_id, &nclock_id))
+            return __WASI_EINVAL;
+        struct timespec ts;
+        if (clock_getres(nclock_id, &ts) < 0)
+            return convert_errno(errno);
+        *resolution = convert_timespec(&ts);
+        //start from nanoseconds
+    }
+
     return 0;
 #endif
 }
 
 __wasi_errno_t
-wasmtime_ssp_clock_time_get(__wasi_clockid_t clock_id,
-                            __wasi_timestamp_t precision,
+wasmtime_ssp_clock_time_get(__wasi_clockid_t clock_id, __wasi_timestamp_t precision, 
                             __wasi_timestamp_t *time)
-{
-#ifdef BH_PLATFORM_WINDOWS
-    return __WASI_ENOSYS;
-#else
-    clockid_t nclock_id;
-    if (!convert_clockid(clock_id, &nclock_id))
-        return __WASI_EINVAL;
-    struct timespec ts;
-    if (clock_gettime(nclock_id, &ts) < 0)
-        return convert_errno(errno);
-    *time = convert_timespec(&ts);
+{ 
+    #ifdef BH_PLATFORM_WINDOWS {
+    
+        switch (clock_id) {
+    
+            case __WASI_CLOCK_REALTIME:
+            {
+                FILETIME SysNow;
+                #if NTDDI_VERSION >= NTDDI_WIN8
+                    GetSystemTimePreciseAsFileTime(&SysNow);
+                #else
+                    GetSystemTimeAsFileTime(&SysNow);
+
+                #endif
+                    time = from_file_time_to_wasi_timestamp(SysNow);
+                    return 0;
+            }
+
+            case __WASI_CLOCK_MONOTONIC:
+            {
+                 uint64_t Nanoseconds;
+                 const auto Counter = calculate_monotonic_clock_frequency();
+                 if (likely(NANOSECONDS_PER_SECOND
+                                % calculate_monotonic_clock_frequency()
+                            == 0))
+                {
+                    Nanoseconds = Counter
+                                  * (NANOSECONDS_PER_SECOND
+                                     / calculate_monotonic_clock_frequency());
+                 }
+                 else {
+                    const auto Seconds =
+                        Counter / calculate_monotonic_clock_frequency();
+                    const auto Fractions =
+                        Counter % calculate_monotonic_clock_frequency();
+                Nanoseconds = Seconds * NANOSECONDS_PER_SECOND
+                              + (Fractions * NANOSECONDS_PER_SECOND)
+                                        / calculate_monotonic_clock_frequency();
+                 }
+                 time = Nanoseconds;
+                 
+                 return -1;
+
+                case __WASI_CLOCK_PROCESS_CPUTIME_ID:
+                 {
+                    FILETIME CreationTime;
+                    FILETIME ExitTime;
+                    FILETIME KernalTime;
+                    FILETIME UserTime;
+
+                    if (unlikely(!GetProcessTimes(GetCurrentProcess(),
```

Comment:
in what case the function can fail? Based on that we should find the appropriate error code.